presto-product-tests/README.md

```diff
@@ -1,125 +1,106 @@
 
-Test assume that you have hadoop and Presto cluster running. 
+*Running product test requires at least 4GB free memory to be available.*
 
-To make it work you need either:
- - define environment variables for hadoop ${HADOOP_MASTER} and Presto ${PRESTO_MASTER} with their IP numbers
- - create test-configuration-local.yaml with following (example content):
+#### On GNU/Linux
+* [```docker >= 1.10```](https://docs.docker.com/installation/#installation)
 
-```
-databases:
-  hive:
-    host: 172.16.2.10
-
-  presto:
-    host: 192.168.205.1
-```
-
-## Running tests
+    ```
+    wget -qO- https://get.docker.com/ | sh
+    ```
 
-Product tests are not run by default. To start them use run following command:
+* [```docker-compose >= 1.60```](https://docs.docker.com/compose/install/)
 
-```
-java -jar target/presto-product-tests-*-executable.jar --config-local tempto-configuration-local.yaml
-```
+    ```
+    pip install docker-compose
+    ```
 
-## Running tests with Using preconfigured docker based clusters
+#### On Mac OS X
 
-In case you do not have an access to hadoop cluster or do not want to spent your time on configuration, this section
-describes how to run product tests with preconfigured docker based clusters.
+* [```docker-toolbox >= 1.10```](https://www.docker.com/products/docker-toolbox)
 
-### Prerequisites
+On Mac OS X installing docker-toolbox gives access to preconfigured bash environment
+with ```docker``` and ```docker-compose``` available. To start this bash environment
+select "Docker Quickstart Terminal" from Launchpad. Note that all commands given in
+further parts of this docs should be run from this environment.
 
-* docker >= 1.10
+##### Setting up virtual machine for docker
 
-[https://docs.docker.com/installation/#installation](https://docs.docker.com/installation/#installation)
+The default ```docker-toolbox``` setting should be decent in most cases, but setting
+up docker virtual machine to have at least 4GB is required.
 
-For linux users:
-```
-wget -qO- https://get.docker.com/ | sh
-```
+* To create virtual machine with name <machine> (required one time):
+    ```
+    docker-machine create -d virtualbox --virtualbox-memory 4096 <machine>
+    ```
+* To set up enviroment to use <machine> (required at docker bash startup):
+    ```
+    eval $(docker-machine env <machine>)
+    ```
 
-For Mac OS X you need to install docker-machine.
+> Tip: In order to keep configuration as simple as possible one may want to change
+> memory available for "default" machine directly in virtual box.
 
-[https://docs.docker.com/engine/installation/mac/](https://docs.docker.com/engine/installation/mac/).
 
-* docker-compose >= 1.6
+## Running tests
 
-[https://docs.docker.com/compose/install/](https://docs.docker.com/compose/install/).
+Product tests are not run by default. To start them use run following command:
 
 ```
-pip install docker-compose
+presto-product-tests/bin/run_on_docker.sh <profile> -x quarantine,big_query
```

Comment:
I would merge the "Configuration profiles" section in here so that you can define <profile> as close as possible to the command that uses it. As it stand right now, a user will look for a definition of <profile> right after the command and won't find one.